cpp/src/randomforest/randomforest.cu

```diff
@@ -695,6 +736,69 @@ RF_metrics score(const raft::handle_t& user_handle,
 /** @} */
 
 // Functions' specializations
+
+/**
+ * @brief Compute feature importances using mean decrease in impurity (internal).
+ */
+template <typename T, typename L>
+static void compute_feature_importances(RandomForestMetaData<T, L>* forest)
+{
+  if (forest->n_features <= 0) {
+    forest->feature_importances_computed = true;
+    return;
+  }
+
+  int n_cols = forest->n_features;
+  std::vector<double> importances(n_cols, 0.0);
+
+  for (const auto& tree : forest->trees) {
+    std::vector<double> tree_importances(n_cols, 0.0);
+
+    if (tree->sparsetree.empty()) continue;
+    double root_sample_count = static_cast<double>(tree->sparsetree[0].InstanceCount());
+
+    if (root_sample_count <= 0) continue;
+
+    for (const auto& node : tree->sparsetree) {
+      if (!node.IsLeaf()) {
+        int feature_id = node.ColumnId();
+        tree_importances[feature_id] += node.BestMetric() * node.InstanceCount();
+      }
+    }
+    for (auto i = 0; i < n_cols; i++) {
+      importances[i] += tree_importances[i] / root_sample_count;
+    }
+  }
+
+  forest->feature_importances.resize(n_cols);
+  double sum = 0.0;
+  for (auto i = 0; i < n_cols; i++) {
+    sum += importances[i];
+  }
+  if (sum > 0) {
+    for (auto i = 0; i < n_cols; i++) {
+      forest->feature_importances[i] = static_cast<T>(importances[i] / sum);
+    }
+  } else {
+    // When all trees are stumps (no splits), explicitly set importances to 0
+    for (auto i = 0; i < n_cols; i++) {
+      forest->feature_importances[i] = static_cast<T>(0);
+    }
+  }
+
+  forest->feature_importances_computed = true;
+}
+
+/**
+ * @brief Get the feature importances of the trained RandomForest model (lazy computation).
+ */
+template <class T, class L>
+std::vector<T> get_feature_importances(RandomForestMetaData<T, L>* forest)
+{
+  if (!forest->feature_importances_computed) { compute_feature_importances(forest); }
+  return forest->feature_importances;
+}
+
```

Comment:
Since the features are persisted in the `RandomForestMetaData` data structure, could we not return a reference to avoid copying?

Reply:
Can we remove the laziness here? For the python API we never expose `RandomForestMetaData`, training returns a treelite model that we directly convert to FIL. I'd like to keep that model, and meaning in practice `get_feature_importances` and the new state stored on `RandomForestMetaData` can go away. `fit_treelite_model` can call `compute_feature_importances` (with an updated signature) to mutate the feature importances ptr inplace before returning.